app/views/bill-runs/match-details.njk

```diff
@@ -0,0 +1,208 @@
+            Charge information
+          </a></li>
+      </ul>
+    </div>
+  </div>
+
+  <div class="govuk-caption-m"> Financial year {{ financialYear }}</div>
+  <h2 class="govuk-heading-m">Charge period {{ chargePeriod }}</h2>
+
+  {# ---------- Billable returns and volume ----------- #}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-full">
+      <div class="govuk-!-margin-right-6 float-left">
+        <h3>
+          <div class="govuk-body govuk-font-size-64 govuk-!-font-weight-bold">
+            {{chargeElement.billableVolume}}ML
+          </div>
+          <div class="govuk-body govuk-!-font-weight-bold govuk-body govuk-!-font-size-19">
+            Billable returns
+          </div>
+        </h3>
+      </div>
+      <div class="govuk-!-margin-right-6 float-left">
+        <h3>
+          <div class="govuk-body govuk-font-size-64 govuk-!-font-weight-bold">
+            /
+          </div>
+        </h3>
+      </div>
+      <div class="govuk-!-margin-right-6 float-left">
+        <h3>
+          <div class="govuk-body govuk-font-size-64 govuk-!-font-weight-bold">
+            {{chargeElement.authorisedVolume}}ML
+          </div>
+          <div class="govuk-body govuk-!-font-weight-bold govuk-body govuk-!-font-size-19">
+            Authorised volume
+          </div>
+        </h3>
+      </div>
+      <div class="govuk-!-margin-right-6 float-left">
+        <h3>
+          {% if chargeElement.issues.length > 0 %}
+            <div class="govuk-inset-text">
+              <h4 class="govuk-heading-s govuk-!-font-weight-bold">
+                Issue
+              </h4>
+              {% for issue in chargeElement.issues %}
+                <div> {{ issue }} </div>
+              {% endfor %}
+            </div>
+          {% endif %}
+        </h3>
+      </div>
+    </div>
+  </div>
+
+  {# ---------- Billable returns button ----------- #}
+  <div class="govuk-grid-row govuk-!-margin-bottom-6">
+    <div class="govuk-grid-column-full">
+      <button class="govuk-button govuk-button--secondary">
+        Edit the billable returns
+      </button>
+    </div>
+  </div>
+
+  {# ------------- Matched Returns --------------- #}
+  {% if matchedReturns.length > 0 %}
+    {% set tableRows = [] %}
+    {% for return in matchedReturns %}
+      {% set matchedReturnIndex = loop.index0 %}
+
+      {% if return.returnStatus == 'overdue' or return.returnStatus == 'query' %}
+        {% set colour = "govuk-tag--red"%}
+        {% set returnLink = "/return/internal?returnId=" + return.returnId %}
+      {% elif return.status == 'void' %}
+        {% set colour = "govuk-tag--grey" %}
+        {% set returnLink = "/return/internal?returnId=" + return.returnId %}
+      {% else %}
+        {% set colour = "govuk-tag--green" %}
+        {% set returnLink = "/returns/return?id=" + return.returnId %}
+      {% endif %}
```

Comment:
We are already working out several other details for each matched return in the presenter. Could we not also determine the `returnLink` in the presenter?

And on determining the color, could that not be dealt with by just using the `badge.njk` macro?